test/bulkhead.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { expect } from 'chai';
```

Comment:
Add tests to cover different behaviors of decorated class method:
- when a method returns a promised value
- when a method throws
- when a method rejects (check original error is rejected)
- when a method is a sync method